lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -245,6 +246,101 @@ def build_authorization_request(money, payment_method, options)
         end
       end
 
+      def add_additional_data(xml, amount, payment_method, options)
+        xml.branchSpecificExtension do
+          xml.purchase do
+            add_level_two_data(xml, amount, payment_method, options)
+            add_level_three_data(xml, amount, payment_method, options)
+          end
+        end
+      end
+
+      def add_level_two_data(xml, amount, payment_method, options)
+        level_2_data = options[:level_two_and_three_data]
```

Comment:
Hi @curiousepic thanks, the idea to use the options[:level_two_and_three_data] is because the fields are basically the same for level 2 and level 3 according this [documentation](https://developer.worldpay.com/docs/wpg/usdomesticacquiring/level23data), therefore I thing it would be ok, or on the other hand maybe we can address each fields as a GSF instead only use level_2_data or level_3_data, but what do you think?

Reply:
I see. Since it is already a fairly well established pattern and naming in AM, I think we should assume that the two separate hashes could be present in options provided to AM, and we can merge them here in the adapter.

Reply:
Ok I will do the change, thanks.